packages/app/src/components/Settings/EditModal.js

```diff
@@ -61,7 +61,17 @@ const EditModal = ({
                         ? new Date(openedOn)
                         : null
                     }
-                    onChange={(value) => handleOpenedOnDateChanges(value)}
+                    onChange={(date) => {
+                      if (!date) {
+                        handleOpenedOnDateChanges(new Date());
+                      } else {
+                        const yearLength = date.getFullYear().toString()
+                          .length;
+                        if (yearLength === 4) {
+                          handleOpenedOnDateChanges(date);
+                        }
```

Comment:
What if instead of converting to string, we just checked if `date.getFullYear()` was between 1000 and 9999?

Reply:
Done - we created a separate function and imported it everywhere we were doing this logic.